### PR TITLE
UI Work for First Pass of Manual Grading

### PR DESCRIPTION
**Pull request by mebird**
- Refactors GET endpoint to use the `async.series` pattern so it's easier to rearrange steps
- Moves the grading interface into a side panel, which acts as an overlay on top of the question interface

### CODE REVIEW

**Review by mwest1066 — APPROVED** [review]
I'm afraid that this will need to be rebased on #3477 once that is edited in response to my recent review. Other than the changes to the underlying #3477, everything in this PR looks good to me.

**Comment by mwest1066** [comment]
@mebird how do you want to proceed with this PR? Do you want to wait updates to the `manual-grading-submission` branch as part of #3477 and then rebase this PR onto the new version, or do you want to merge this into the `manual-grading-submission` branch now? Either is fine with me.

**Comment by mebird** [comment]
> @mebird how do you want to proceed with this PR? Do you want to wait updates to the `manual-grading-submission` branch as part of #3477 and then rebase this PR onto the new version, or do you want to merge this into the `manual-grading-submission` branch now? Either is fine with me.

There's one or two issues I'm debugging locally with missing submission IDs, so this PR is going to need another day and a bit. I rebase this PR onto the new version, since it's likely some of the async code I touched is going to change.

**Comment by mwest1066** [comment]
> There's one or two issues I'm debugging locally with missing submission IDs, so this PR is going to need another day and a bit. I rebase this PR onto the new version, since it's likely some of the async code I touched is going to change.

Sounds great! I should be able to turn around code reviews and merges very rapidly while we are in winter break (until the end of January), so keep them coming!

File: docs/docker.md [hunk 1]
```diff
@@ -72,6 +72,51 @@ E.g., to start a shell in a container started with `--name pl`:
 docker exec -it pl /bin/bash
 ```
 
+## Docker-Compose
+
+### Basics
+
+A docker-compose file describes the services an application needs to run. In our case, we use `docker-compose` to configure and run the PrairieLearn docker container locally.
+
+To run PrairieLearn with `docker-compose`, run `docker-compose up pl`. This will, in order:
```
Comment (andrewstec) [inline docs/docker.md:81]: You can use a numbered list to denote the sequential nature of the steps.

Comment (andrewstec) [reply docs/docker.md:81]: I don't know if search engines read markdown syntax for semantics yet, but I wouldn't be surprised if they do. They do read semantics in HTML.

**Comment by mebird** [comment]
Pointing towards master instead.

**Pull request closed without merging**